File: Client/Android/component/SolutionBase/src/main/java/com/vertcdemo/core/net/IRequestCallback.java
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0

package com.vertcdemo.core.net;

public interface IRequestCallback<T> {

    void onSuccess(T data);

    default void onError(int errorCode, String message) {
    }
}

File: Client/Android/component/solution-base/src/main/java/com/vertcdemo/core/net/IRequestCallback.kt
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0
package com.vertcdemo.core.net

@JvmDefaultWithCompatibility
fun interface IRequestCallback<T> {
    fun onSuccess(data: T)

    fun onError(errorCode: Int, message: String?) {
    }
}
